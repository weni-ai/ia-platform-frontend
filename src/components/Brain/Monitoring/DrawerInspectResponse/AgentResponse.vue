<template>
  <section class="inspect-response__agent">
    <UnnnicIntelligenceText
      data-testid="agent-response-title"
      color="neutral-cloudy"
      family="secondary"
      size="body-gt"
      tag="p"
    >
      {{ $t('router.monitoring.inspect_response.agent_response') }}:
    </UnnnicIntelligenceText>
    <ResponseSuccessGroundedness
      v-if="showSuccessGroundedness"
      data-testid="agent-response-success-groundedness"
      :inspectionData="inspectionData"
    />
    <p
      v-else
      data-testid="agent-response-text"
      class="agent__text"
    >
      “{{ response }}”
    </p>
  </section>
</template>

<script setup>
import { computed } from 'vue';

import ResponseSuccessGroundedness from '../ResponseSuccessGroundedness/index.vue';

const props = defineProps({
  status: {
    type: String,
    required: true,
  },
  response: {
    type: String,
    default: '',
  },
  inspectionData: {
    type: Object,
    required: true,
  },
});

const showSuccessGroundedness = computed(
  () =>
    props.status === 'success' &&
    props.inspectionData.groundedness.find(
      (fragment) => fragment.sources.length,
    ),
);
</script>
