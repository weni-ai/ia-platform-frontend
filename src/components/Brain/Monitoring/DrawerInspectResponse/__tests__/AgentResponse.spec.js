import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';
import AgentResponse from '../AgentResponse.vue';

describe('AgentResponse.vue', () => {
  let wrapper;

  const defaultProps = {
    status: 'error',
    response: 'Sample agent response',
    inspectionData: { key: 'value' },
  };

  const mountWrapper = (props = {}) => {
    return shallowMount(AgentResponse, {
      props: {
        ...defaultProps,
        ...props,
      },
    });
  };

  beforeEach(() => {
    wrapper = mountWrapper();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the agent response with correct props', () => {
    const agentResponseTitle = wrapper.findComponent(
      '[data-testid="agent-response-title"]',
    );
    expect(agentResponseTitle.exists()).toBe(true);
    expect(agentResponseTitle.props()).toMatchObject({
      color: 'neutral-cloudy',
      family: 'secondary',
      size: 'body-gt',
      tag: 'p',
    });
  });

  it('renders the agent response text when status is not "success"', () => {
    const agentResponse = wrapper.find('[data-testid="agent-response-text"]');
    expect(agentResponse.exists()).toBe(true);
    expect(agentResponse.text()).toBe(`“${defaultProps.response}”`);
  });

  it('does not render groundedness when status is not "success"', () => {
    const groundedness = wrapper.findComponent(
      '[data-testid="agent-response-success-groundedness"]',
    );
    expect(groundedness.exists()).toBe(false);
  });

  it('renders the agent response text when groundedness sources are empty', () => {
    const inspectionData = {
      groundedness: [{ sources: [] }, { sources: [] }],
    };
    wrapper = mountWrapper({ status: 'success', inspectionData });
    const agentResponse = wrapper.find('[data-testid="agent-response-text"]');
    expect(agentResponse.exists()).toBe(true);
    expect(agentResponse.text()).toBe(`“${defaultProps.response}”`);
  });

  it('renders groundedness when status is "success" and sources are not empty', () => {
    const inspectionData = {
      groundedness: [{ sources: ['source1'] }, { sources: [] }],
    };
    wrapper = mountWrapper({ status: 'success', inspectionData });
    const groundedness = wrapper.findComponent(
      '[data-testid="agent-response-success-groundedness"]',
    );
    expect(groundedness.exists()).toBe(true);
    expect(groundedness.props()).toMatchObject({
      inspectionData,
    });
  });

  it('does not render the response text when status is "success" and sources are not empty', () => {
    const inspectionData = {
      groundedness: [{ sources: ['source1'] }],
    };
    wrapper = mountWrapper({ status: 'success', inspectionData });
    const responseText = wrapper.find('[data-testid="agent-response-text"]');
    expect(responseText.exists()).toBe(false);
  });

  it('applies the correct class to the section', () => {
    const section = wrapper.find('section');
    expect(section.classes()).toContain('inspect-response__agent');
  });
});
