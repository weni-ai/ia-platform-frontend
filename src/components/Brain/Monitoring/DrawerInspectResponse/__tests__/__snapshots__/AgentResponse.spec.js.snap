// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AgentResponse.vue > matches the snapshot 1`] = `
"<section class="inspect-response__agent">
  <unnnic-intelligence-text-stub size="body-gt" tag="p" family="secondary" lineheight="md" weight="regular" color="neutral-cloudy" data-testid="agent-response-title"></unnnic-intelligence-text-stub>
  <p data-testid="agent-response-text" class="agent__text"> “Sample agent response” </p>
</section>"
`;
